lib/rules/no-broken-super-chain.js

```diff
@@ -3,93 +3,107 @@
+    isSuperCall = lineWithinFn.object.property.name === '_super';
+  }
+
+  return isSuperCall;
+}
 
-    if (node.parent.parent
-      && node.parent.parent.parent
-      && node.parent.parent.parent.callee
-      && node.parent.parent.parent.callee.object
-      && node.parent.parent.parent.callee.object.object
-      && node.parent.parent.parent.callee.object.object.name === 'Ember') {
-      return (node.parent.parent.parent.callee.object.property
-        && EMBER_MODULES_WITH_SUPER_CHAIN[node.parent.parent.parent.callee.object.property.name]);
-    }
+function wereArgumentsPassedToSuper(expression) {
+  const callee = expression.callee;
+
+  if (!expression || callee.type !== 'MemberExpression') {
+    return;
   }
 
-  return false;
+  if (callee.property.name === '_super') {
+    const firstArgumentToSuper = expression.arguments[0];
+    return firstArgumentToSuper && firstArgumentToSuper.type === 'SpreadElement' && firstArgumentToSuper.argument.name === 'arguments';
+  } else if (callee.property.name === 'apply') {
+    const args = expression.arguments;
+    return args.length >= 2 && args[0].type === 'ThisExpression' && args[1].name === 'arguments';
+  }
 }
 
 module.exports = {
   meta: {
     docs: {
-      description: 'Prevent the absence of `this._super(...arguments)` in `init()` calls or the use of `this` prior to `this._super()`',
+      description: 'Prevent the absence of `this._super(...arguments)` in calls to various lifecycle hooks or the use of `this` prior to `this._super()`',
       category: 'Best Practices',
       recommended: true
     },
     messages: MESSAGES
   },
   create(context) {
-    let initOverride = null;
-
     return {
-      onCodePathStart(codePath, node) {
-        if (isInit(node)) {
-          initOverride = {
-            superCalled: false,
-            superCalledFirst: false
-          };
-        }
-      },
-      onCodePathEnd(codePath, node) {
-        if (initOverride && isInit(node)) { // TODO: Maybe check against codepath.name
-          if (!initOverride.superCalled) {
-            context.report({
-              message: MESSAGES.noSuper,
-              node
-            });
-          }
+      CallExpression(node) {
+        if (isExtend(node)) {
+          let superCount = 0;
+          const extendedObjects = node.arguments;
 
-          initOverride = null;
-        }
-        return;
-      },
-      'CallExpression:exit'(node) {
-        if (initOverride) {
-          const property = node.callee.property;
-          if (property && property.type === 'Identifier' && property.name === '_super') {
-            if (initOverride.superCalled) {
-              context.report({
-                message: MESSAGES.tooManySupers,
-                node
-              });
-            } else {
-              initOverride.superCalled = true;
-            }
-          }
+          extendedObjects.forEach(extendedObj => {
+            extendedObj.properties.forEach(property => {
```

Comment:
We need to guard for `extendedObj.properties` being undefined here (or filter out non `ObjectExpression`s before `.forEach`ing).

I believe the following test should be added (which will fail ATM but pass if we filtered non ObjectExpressions):

```js
Foo.extend(SomeMixin, {
  init() {
    this._super(...arguments);
  }
});
```

I'd suggest either of the following:

```js
extendedObjects = node.arguments.filter(arg => arg.type === 'ObjectExpression');

// or

extendedObjects.forEach(extendedObj => {
  if (extendedObj.type !== 'ObjectExpression') { return; }

  // ...snip...
});
```

Reply:
I would also suggest adding a test case for this.

Reply:
Updated, and added test.